columnflow/tasks/framework/histograms.py

```diff
@@ -0,0 +1,192 @@
+# coding: utf-8
+
+"""
+Base tasks for working with multiple merged histograms.
+"""
+
+from __future__ import annotations
+
+import law
+import order as od
+
+from columnflow.tasks.framework.base import Requirements, ShiftTask
+from columnflow.tasks.framework.mixins import (
+    CalibratorsMixin, SelectorStepsMixin, ProducersMixin, MLModelsMixin, WeightProducerMixin,
+    VariablesMixin, DatasetsProcessesMixin, CategoriesMixin,
+    ShiftSourcesMixin,
+)
+from columnflow.tasks.histograms import MergeHistograms, MergeShiftedHistograms
+from columnflow.util import dev_sandbox, maybe_import
+
+ak = maybe_import("awkward")
+hist = maybe_import("hist")
+
+
+class HistogramsUserBase(
+    DatasetsProcessesMixin,
+    CategoriesMixin,
+    VariablesMixin,
+    MLModelsMixin,
+    WeightProducerMixin,
+    ProducersMixin,
+    SelectorStepsMixin,
+    CalibratorsMixin,
+):
+    sandbox = dev_sandbox(law.config.get("analysis", "default_columnar_sandbox"))
+
+    def store_parts(self):
+        parts = super().store_parts()
+        parts.insert_before("version", "datasets", f"datasets_{self.datasets_repr}")
+        return parts
+
+    def load_histogram(
+        self,
+        dataset: str | od.Dataset,
+        variable: str | od.Variable,
+    ) -> hist.Hist:
+        """
+        Helper function to load the histogram from the input for a given dataset and variable.
+
+        :param dataset: The dataset name or instance.
+        :param variable: The variable name or instance.
+        :return: The loaded histogram.
+        """
+        if isinstance(dataset, od.Dataset):
+            dataset = dataset.name
+        if isinstance(variable, od.Variable):
+            variable = variable.name
+        histogram = self.input()[dataset]["collection"][0]["hists"].targets[variable].load(formatter="pickle")
+        return histogram
+
+    def slice_histogram(
+        self,
+        histogram: hist.Hist,
+        processes: str | list[str],
+        categories: str | list[str],
+        shifts: str | list[str],
+        reduce_axes: bool = False,
+    ) -> hist.Hist:
+        """
+        Slice a histogram by processes, categories, and shifts.
+
+        This function takes a histogram and slices it based on the specified processes, categories, and shifts.
+        It returns the sliced histogram.
```

Comment:
Hm, I'm not sure how familiar users are with the concept of 'slicing a histogram' (I for one am not sure if I understand it correctly :D). Maybe we can be slightly more specific here:

```suggestion
        Slice a histogram by processes, categories, and shifts.

        This function takes a histogram and slices it based on the specified processes, categories, and shifts, i.e. all  other axes are integrated out.
        It returns the sliced histogram.
```